direct-query-core/src/main/java/org/opensearch/sql/directquery/DirectQueryExecutorServiceImpl.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.sql.directquery;
+
+import java.io.IOException;
+import java.util.UUID;
+import org.opensearch.common.inject.Inject;
+import org.opensearch.sql.datasource.client.DataSourceClientFactory;
+import org.opensearch.sql.datasource.client.exceptions.DataSourceClientException;
+import org.opensearch.sql.datasource.query.QueryHandlerRegistry;
+import org.opensearch.sql.directquery.rest.model.ExecuteDirectQueryRequest;
+import org.opensearch.sql.directquery.rest.model.ExecuteDirectQueryResponse;
+import org.opensearch.sql.directquery.rest.model.GetDirectQueryResourcesRequest;
+import org.opensearch.sql.directquery.rest.model.GetDirectQueryResourcesResponse;
+
+public class DirectQueryExecutorServiceImpl implements DirectQueryExecutorService {
+
+  private final DataSourceClientFactory dataSourceClientFactory;
+  private final QueryHandlerRegistry queryHandlerRegistry;
+
+  @Inject
+  public DirectQueryExecutorServiceImpl(
+      DataSourceClientFactory dataSourceClientFactory, QueryHandlerRegistry queryHandlerRegistry) {
+    this.dataSourceClientFactory = dataSourceClientFactory;
+    this.queryHandlerRegistry = queryHandlerRegistry;
+  }
+
+  @Override
+  public ExecuteDirectQueryResponse executeDirectQuery(ExecuteDirectQueryRequest request) {
+    // TODO: Replace with the data source query id.
+    String queryId = UUID.randomUUID().toString();
+    String sessionId = request.getSessionId(); // Session ID is passed as is
+    String dataSourceName = request.getDataSources();
+    String dataSourceType = null;
+    String result;
+
+    try {
+      // Get data source type before creating the client
+      dataSourceType =
+          dataSourceClientFactory.getDataSourceType(dataSourceName).name().toLowerCase();
+
+      // Let Java infer the type
+      var client = dataSourceClientFactory.createClient(dataSourceName);
+
+      // No need to specify Object type parameter, let the registry infer it
+      result =
+          queryHandlerRegistry
+              .getQueryHandler(client)
+              .map(
+                  handler -> {
+                    try {
+                      return handler.executeQuery(client, request);
+                    } catch (IOException e) {
+                      return "{\"error\": \"Error executing query: " + e.getMessage() + "\"}";
```

Comment:
if prometheus returns an error, should the SQL request fail with non-200 response?